### DISCUSSION TITLE
Plugin i18n

### DISCUSSION THREAD

**Discussion by mpaglia0** [getpelican/pelican#3272] [Ideas]
Hi all,
I need to add i18n to a plugin that writes a short string in a post.
My idea is to use `gettext`.
Do you think it is acceptable to use a `locale/<language>/LC_MESSAGES` structure starting from the `plugins` directory?
In this way any plugins can have access to .po files via the `../locale` path.
Directory structure will be the following:
`plugins/locale/it/LC_MESSAGES/<plugin_name>.po` etc.

Thank you,
Maurizio

**Comment by avaris** [comment]
I would suggest keeping plugin files inside plugin folder.

**Reply by mpaglia0** [reply]
Yes, this is my question/proposal.
The final directory structure could be the following:
![IMAGE](https://github.com/getpelican/pelican/assets/21242068/6fd349d0-5184-4570-ae6e-3a7c8c35919d)
I am asking in order to know if mantainers can confirm a structure that can be kept valid for all plugins i18n

Thank you!!
Maurizio

**Reply by justinmayer** [reply]
I believe @avaris was suggesting: `plugins/plugin_01/locale/it_IT/LC_MESSAGES/plugin_01.mo`

What are the benefits of centralizing all plugin localization files in a single directory? I imagine there must be some benefits, which is why you suggested that layout, right? 😊

**Reply by avaris** [reply]
> I believe @avaris was suggesting: plugins/plugin_01/locale/it_IT/LC_MESSAGES/plugin_01.mo

Yes, sorry if my initial comment was confusing. Bleeding things outside your own plugin folder can become messy and introduce more points of failure. Because now, you'll be relying on other plugins to do the "right thing". And I don't see any obvious benefits to that.

Plus, your structure effectively makes `locale` a reserved name and no one can make a plugin named `locale`.

**Reply by mpaglia0** [reply]
@justinmayer , @avaris please apologize if I misunderstood.
I thought a common tree could be better in order to have an 'official' Pelican structure.
I understand your point of view so I will go on with @avaris  suggestion i.e. `plugins/<my_plugin>/locale/<languages>/LC_MESSAGES`
Thank you indeed both for your time!

Have a nice day,
Maurizio
